### ISSUE TITLE
Generate errors for missing values

### ISSUE THREAD

**Issue by saulakravitz** [nightingaleproject/vrdr-dotnet#364]
One of the common pathologies observed in jurisdiction submissions are Observations that don't contain any value.

Code should be added to detect these situations (both the Observation.value and the Observation.component.value) and to generate an error.

**Comment by johnarleyburns** [comment]
It is not clear what is considered an error situation.  Please provide additional guidance on when Observation must have a value element and when it cannot.  For instance, many times a valueCodeableConcept may be present, is this considered a value or not.

**Comment by saulakravitz** [comment]
I amended the issue with some examples.    But the IG is rife with similar instances of required values, extension values, or component values.

**Comment by johnarleyburns** [comment]
Should we add a test for each of the Business Rules from the business rule document, is that what you are asking?

**Comment by saulakravitz** [comment]
No.  For each profile with a required Value.   Business rules are not maintained in the VRDR .NET library which is focused on VRDR.

**Comment by consultantusa** [comment]
- Search for each of Observation derivatives from VRDR on IG to check on Observation.value or Observation.component.value status to see if it is required
- For those Observation derivatives with required values, check on their values programmatically and generate error if they are missing.

**Comment by whichwit** [comment]
`Observation.value` can be identified as required or optional. `Observation.component.value` is not the interest here, but the component itself may be required to an `Observation`.

Error should generate only when processing a resource would make the death record non-conforming to the VRDR IG.

Checking for required `Observation.component` may be a business rule implementation that is better handled in Canary.

**Comment by consultantusa** [comment]
"Error should generate only when processing a resource would make the death record non-conforming to the VRDR IG"

Is missing value, that is required per VRDR IG, in VRDR .NET app considered non-conforming? In other words, how it it different from "For those Observation derivatives with required values, check on their values programmatically and generate error if they are missing"?

**Comment by whichwit** [comment]
If you don't see the difference between our comments, perhaps it will be helpful to elaborate a bit what you meant by "for those Observation derivatives with required values, check on their values programmatically and generate error if they are missing." and how does doing so help to resolves this request?

**Comment by consultantusa** [comment]
Updated list with required Observation.value and/or required its component and/or required its component's value:

Birth Record Identifier => BirthRecordIdentifier/CreateBirthRecordIdentifier // Observation.component:birthJurisdiction.value[x] required

Decedent Age => AgeAtDeathObs/CreateAgeAtDeathObs //

Decedent Education Level => DecedentEducationLevel/CreateEducationLevelObs // Observation.value[x] required

Decedent Military Service => MilitaryServiceObs/RestoreReferences/properties // Observation.value[x] required

Decedent Usual Work => UsualWork/CreateUsualWork // component:odh-UsualIndustry required

Emerging Issues => EmergingIssues/RestoreReferences/properties //

Input Race and Ethnicity => InputRaceAndEthnicityObs/CreateInputRaceEthnicityObs // component.value required

Autopsy Performed Indicator => AutopsyPerformed/CreateAutopsyPerformed // Observation.value[x]	required

Death Date => DeathDateObs/CreateDeathDateObs //

Decedent Pregnancy Status => PregnancyObs/CreatePregnancyObs // Observation.value[x] required

Examiner Contacted => ExaminerContactedObs/RestoreReferences/properties // Observation.value[x] required

Injury Incident => InjuryIncidentObs/CreateInjuryIncidentObs //

Surgery Date => SurgeryDateObs/CreateSurgeryDateObs //

Tobacco Use Contributed To Death => TobaccoUseObs/RestoreReferences // Observation.value[x] required

Cause Of Death Part 1 => CodCondition/CauseOfDeathCondition/properties // Observation.value[x] required, component:lineNumber required

Cause Of Death Part 2 => CodCondition/CauseOfDeathCondition/properties // Observation.value[x] required

Manner of Death => MannerOfDeath/GetCauseOfDeathCodedContentBundle/RestoreReferences //  Observation.value[x] required

Decedent Disposition Method => DispositionMethod/RestoreReferences/DecedentDispositionMethod // Observation.value[x] required

Activity at Time of Death => ActivityAtTimeOfDeathObs/CreateActivityAtTimeOfDeathObs // Observation.value[x] required

Automated Underlying Cause Of Death => AutomatedUnderlyingCauseOfDeathObs/CreateAutomatedUnderlyingCauseOfDeathObs

Coded Race and Ethnicity => CodedRaceAndEthnicityObs/CreateCodedRaceAndEthnicityObs // component:First-EighthEditedCode.value required, component:First-SecondAmericanIndianCode.value required, component:First-SecondOtherAsianCode.value required, component:First-SecondOtherPacificIslanderCode.value required, component:First-SecondOtherRaceCode.value required, component:RaceRecode40.value requierd, component:HispanicCode.value required, component:HispanicCodeForLiteral.value required 

Entity Axis Cause Of Death => EntityAxisCauseOfDeathObsList/properties // Observation.value[x] required, component:lineNumber/value required, component:position/value required

Manual Underlying Cause Of Death => ManualUnderlyingCauseOfDeathObs/CreateManualUnderlyingCauseOfDeathObs // Observation.value[x] required

Place Of Injury => PlaceOfInjuryObs/CreatePlaceOfInjuryObs // Observation.value[x] required

Record Axis Cause Of Death => RecordAxisCauseOfDeathObsList/properties // Observation.value[x] required

**Comment by whichwit** [comment]
This is a list we can start with. Please move forward with implementation on requiring `Observation.value` for Decedent Military Service in the .NET library, along with unit tests. Checking for `Observation.Component` should be done in Canary as previously reviewed.

The same logic will be applied to multiple resource profiles, please consider code reuse. Do **not** move beyond Decedent Military Service until we have a chance to review the code and unit tests.

**Comment by consultantusa** [comment]
That linked PR was NOT created by me on my work IN-PROGRESS, which might not need to proceed at all as the issue was likely already addressed, along with its accompanied test, as I have discussed.

**Comment by consultantusa** [comment]
The recent commit covers updates on codes and tests, for both parsed data and manual input data, for MilitaryService object.
